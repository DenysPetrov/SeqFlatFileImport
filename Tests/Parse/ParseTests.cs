using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using SeqFlatFileImport.Tests.Helpers;

namespace SeqFlatFileImport.Tests.Parse
{
    public class ParseTests
    {
        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusServer()
        {
            Execute("OctopusServer.txt");
        }

        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusWebLog()
        {
            Execute("Web-2017-01-18.log");
        }


        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Iis()
        {
            Execute("u_ex170117.log");
        }

        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusTask()
        {
            Execute("ServerTasks-16572.log.txt");
        }

        [Test]
        [UseReporter(typeof (BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusRawTask()
        {
            Execute("servertasks-21_gjzwyyv2kt.txt");
        }

        public void Execute(string inputFileName)
        {
            var seqServer = new StubSeqEndpoint();
            var result = new Importer(seqServer)
                .Import(TestHelper.GetFilePath(inputFileName));
            result.ShouldBeSuccessful();
            Approvals.VerifyJson(seqServer.LogsAsJson);
        }


    }
}