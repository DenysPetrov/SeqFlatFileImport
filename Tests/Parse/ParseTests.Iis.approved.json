[
    {
        "Timestamp": "2017-01-17T01:41:39+00:00",
        "Level": "Information",
        "MessageTemplate": "{cs_method} {time_taken}{sc_status}.{sc_substatus} {cs_uri_stem}  {cs_uri_query}",
        "Properties": {
            "date": "2017-01-17",
            "time": "01:41:39",
            "s_ip": "::1",
            "cs_method": "GET",
            "cs_uri_stem": "/",
            "cs_uri_query": "-",
            "s_port": "80",
            "cs_username": "-",
            "c_ip": "::1",
            "cs(User_Agent)": "Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/55.0.2883.87+Safari/537.36",
            "cs(Referer)": "-",
            "sc_status": "200",
            "sc_substatus": "0",
            "sc_win32_status": "64",
            "time_taken": "576"
        },
        "Exception": null
    },
    {
        "Timestamp": "2017-01-17T01:41:39+00:00",
        "Level": "Information",
        "MessageTemplate": "{cs_method} {time_taken}{sc_status}.{sc_substatus} {cs_uri_stem}  {cs_uri_query}",
        "Properties": {
            "date": "2017-01-17",
            "time": "01:41:39",
            "s_ip": "::1",
            "cs_method": "GET",
            "cs_uri_stem": "/",
            "cs_uri_query": "-",
            "s_port": "80",
            "cs_username": "-",
            "c_ip": "::1",
            "cs(User_Agent)": "Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/55.0.2883.87+Safari/537.36",
            "cs(Referer)": "-",
            "sc_status": "200",
            "sc_substatus": "0",
            "sc_win32_status": "0",
            "time_taken": "1"
        },
        "Exception": null
    },
    {
        "Timestamp": "2017-01-17T01:41:39+00:00",
        "Level": "Information",
        "MessageTemplate": "{cs_method} {time_taken}{sc_status}.{sc_substatus} {cs_uri_stem}  {cs_uri_query}",
        "Properties": {
            "date": "2017-01-17",
            "time": "01:41:39",
            "s_ip": "::1",
            "cs_method": "GET",
            "cs_uri_stem": "/iisstart.png",
            "cs_uri_query": "-",
            "s_port": "80",
            "cs_username": "-",
            "c_ip": "::1",
            "cs(User_Agent)": "Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/55.0.2883.87+Safari/537.36",
            "cs(Referer)": "http://localhost/",
            "sc_status": "304",
            "sc_substatus": "0",
            "sc_win32_status": "0",
            "time_taken": "1"
        },
        "Exception": null
    }
]